Revit_Core_Engine/Modify/SortBoundaryLoops.cs

```diff
@@ -0,0 +1,78 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2024, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using Autodesk.Revit.DB;
+using BH.oM.Base.Attributes;
+using System.Collections.Generic;
+using System.ComponentModel;
+
+namespace BH.Revit.Engine.Core
+{
+    public static partial class Modify
+    {
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        [Description("Sort a list of curve loops (from a spatial element) so that the first one corresponds to the main loop that contains all others.")]
+        [Input("loops", "Boundary curve loops from a spatial element like a room or space.")]
+        [Output("normal", "Normal vector of the surface the curve loops come from, usually come from face.ComputeNormal or face.FaceNormal if it's a planar face.")]
+        public static void SortBoundaryLoops(this List<CurveLoop> loops, XYZ normal)
+        {
+            if (loops.Count < 2)
+                return;
+
+            //By convention, the main boundary must be counter-clockwise relative to the normal vector.
+            while (loops[0].IsCounterclockwise(normal) == false)
```

Comment:
Couldn't we find the index of first counterclockwise curve in set with `FindIndex`, then remove at that index and insert the curve at 0?